File: aeris/schemas/rag.py
from typing import Optional, List
from pydantic import BaseModel, Field


class KnowledgeBaseCreate(BaseModel):
    """创建知识库请求"""
    name: str = Field(..., max_length=100)
    description: str = Field(default="", max_length=500)


class KnowledgeBaseResponse(BaseModel):
    """知识库响应"""
    id: int
    name: str
    description: str
    collection_name: str
    is_active: bool
    created_at: str
    updated_at: Optional[str] = None

    class Config:
        from_attributes = True


class DocumentUploadResponse(BaseModel):
    """文档上传响应"""
    id: int
    knowledge_base_id: int
    title: str
    source_type: str
    source_path: str
    status: str
    created_at: str


class URLFetchRequest(BaseModel):
    """URL 抓取请求"""
    url: str = Field(..., max_length=500)


class URLFetchResponse(BaseModel):
    """URL 抓取响应"""
    id: int
    knowledge_base_id: int
    title: str
    source_type: str
    source_path: str
    status: str
    created_at: str


class DocumentStatusResponse(BaseModel):
    """文档状态响应"""
    id: int
    status: str
    chunk_count: int
    error_message: Optional[str] = None


class SearchRequest(BaseModel):
    """检索请求"""
    knowledge_base_ids: Optional[List[int]] = None
    query: str = Field(..., max_length=1000)
    top_k: int = Field(default=5, ge=1, le=20)


class SearchResultItem(BaseModel):
    """检索结果项"""
    kb_id: int
    kb_name: str
    document_id: int
    chunk_id: str
    content: str
    score: float


class SearchResponse(BaseModel):
    """检索响应"""
    results: List[SearchResultItem]

File: frontend/src/services/rag.ts
import api from './api'

export interface KnowledgeBase {
  id: number
  name: string
  is_active: boolean
  created_at: string
  updated_at: string | null
  document_count: number
  chunk_count: number
}

export interface Document {
  id: number
  knowledge_base_id: number
  title: string
  source_type: string
  source_path: string
  status: string
  chunk_count: number
  created_at: string
  updated_at: string | null
}

export interface KnowledgeBaseDetail {
  id: number
  name: string
  is_active: boolean
  created_at: string
  updated_at: string | null
  document_count: number
  chunk_count: number
  documents: Document[]
}

export const ragApi = {
  getKnowledgeBases() {
    return api.get<KnowledgeBase[]>('/rag/kb')
  },
  getKnowledgeBaseDetail(kbId: number) {
    return api.get<KnowledgeBaseDetail>(`/rag/kb/${kbId}`)
  },
}